zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -3398,6 +3403,26 @@ private ClientCnxnSocket getClientCnxnSocket() throws IOException {
         }
     }
 
+    /**
+     * Return the default watcher of this instance if requested.
+     *
+     * @param requested if the default watcher requested
+     * @return the default watcher if request, otherwise {@code null}.
+     * @throws IllegalStateException if a null default watcher is requested
+     */
+    @Nullable
```

Comment:
I am not sure about this.
Are we using it in ZK?

Reply:
I do a global check and find there isn't anywhere else using it. Possibly annotate `@Nullable` is a good idea which helps notify users taking care of nullable value but let's discuss in another thread.

In short, reverted.